docs/source/howto/run_codes.rst

```diff
@@ -585,6 +585,33 @@ Finally, the monitor needs to be declared using an entry point in the ``aiida.ca
 The next section will show how this entry point is used to assign it to a calculation job.
 
 
+.. versionadded:: 2.5.0
+
+Monitors can also attach outputs to the calculation that it is monitoring.
+This can be useful to report outputs while the calculation is running that should still be stored permanently in the provenance graph.
```

Comment:
At least in the use case of Aurora, there is no desire to store snapshots permanently. In the current implementation, I continuously overwrite `extras.snapshot` with the latest. The field is discarded on job completion. But perhaps partial data can be important in some cases, say, in a structure optimization calculation, where intermediates are of value down the road (though many if not all codes take care of this internally).